File: WebContent/js/controllers.js
var webControllers = angular.module('webControllers', []);

webControllers.controller('sampleCtrl', [ '$scope', '$http', function($scope, $http) {
	
	$scope.sampleVar = "Hello World!!";
	$http.get('/AttendancePortal/rest/sampleRestApi').success(function(data) {
		  console.log("Profile data"+ data);  
		  $scope.restCallData = data;
		});
}]);

webControllers.controller('addFacultyCtrl', [ '$scope', '$http', function($scope, $http) {
	$scope.errorFlag = false;
	$scope.successFlag = false;
    $scope.addFaculty = function(){
        $scope.errorFlag = false;
        $http.get('/AttendancePortal/rest/addFacultyDo', {params:{firstName:$scope.firstName, lastName:$scope.lastName, id:$scope.id, phoneNumber:$scope.phoneNumber, password:$scope.password}}).success(function(data) {
            console.log("Got data from server")
            $scope.response = data;
            if(data == false) {
            	$scope.errorFlag = true;
            	$scope.successFlag = false;
            } else {
            	$scope.successFlag = true;
            }
          }).error(function(){
              console.log("No reply from server");
              $scope.errorFlag = true;
              $scope.successFlag = false;
            });

    	$scope.firstName = null;
    	$scope.lastName = null;
    	$scope.id = null;
    	$scope.phoneNumber = null;
    	$scope.password = null;
    }
}]);

webControllers.controller('addStudentCtrl', [ '$scope', '$http', function($scope, $http) {
	$scope.errorFlag = false;
	$scope.successFlag = false;
    $scope.addFaculty = function(){
        $scope.errorFlag = false;
        $http.get('/AttendancePortal/rest/addStudentDo', {params:{firstName:$scope.firstName, lastName:$scope.lastName, fathersName:$scope.fathersName, branch:$scope.branch, year:$scope.year, id:$scope.id, phoneNumber:$scope.phoneNumber, password:$scope.password, courses:$scope.courses}}).success(function(data) {
            console.log("Got data from server")
            $scope.response = data;
            if(data == false) {
            	$scope.errorFlag = true;
            	$scope.successFlag = false;
            } else {
            	$scope.successFlag = true;
            }
          }).error(function(){
              console.log("No reply from server");
              $scope.errorFlag = true;
              $scope.successFlag = false;
            });

    	$scope.firstName = null;
    	$scope.lastName = null;
    	$scope.fathersName = null;
    	$scope.branch = null;
    	$scope.id = null;
    	$scope.phoneNumber = null;
    	$scope.password = null;
    	$scope.year = null;
    	$scope.courses = null;
    }
}]);

webControllers.controller('adminCtrl', [ '$scope', '$http', function($scope, $http) {
	
}]);

webControllers.controller('deleteFacultyCtrl', [ '$scope', '$http', function($scope, $http) {
	$scope.errorFlag = false;
	$scope.successFlag = false;
    $scope.deleteFaculty = function(){
        $scope.errorFlag = false;
        $http.get('/AttendancePortal/rest/deleteFacultyDo', {params:{id:$scope.id}}).success(function(data) {
            console.log("Got data from server")
            $scope.response = data;
            if(data == false) {
            	$scope.errorFlag = true;
            	$scope.successFlag = false;
            } else {
            	$scope.successFlag = true;
            }
          }).error(function(){
              console.log("No reply from server");
              $scope.errorFlag = true;
              $scope.successFlag = false;
            });
    	$scope.id = null;
    }
}]);

webControllers.controller('deleteStudentCtrl', [ '$scope', '$http', function($scope, $http) {
	$scope.errorFlag = false;
	$scope.successFlag = false;
    $scope.deleteStudent = function(){
        $scope.errorFlag = false;
        $http.get('/AttendancePortal/rest/deleteStudentDo', {params:{id:$scope.id}}).success(function(data) {
            console.log("Got data from server")
            $scope.response = data;
            if(data == false) {
            	$scope.errorFlag = true;
            	$scope.successFlag = false;
            } else {
            	$scope.successFlag = true;
            }
          }).error(function(){
              console.log("No reply from server");
              $scope.errorFlag = true;
              $scope.successFlag = false;
            });
    	$scope.id = null;
    }
}]);

webControllers.controller('enterAttendanceCtrl', [ '$scope', '$http', '$location', function($scope, $http, $location) {
	$scope.id = $location.path().split("/")[2];
	$scope.errorFlag = false;
	$scope.successFlag = false;
	$scope.studentList = "";
    $scope.enterAttendance = function(){
        $scope.errorFlag = false;
        $http.get('/AttendancePortal/rest/enterAttendanceDo', {params:{courseId:$scope.courseId, date:$scope.date, studentList:$scope.studentList}}).success(function(data) {
            console.log("Got data from server")
            $scope.response = data;
            if(data == false) {
            	$scope.errorFlag = true;
            	$scope.successFlag = false;
            } else {
            	$scope.successFlag = true;
            }
          }).error(function(){
              console.log("No reply from server");
              $scope.errorFlag = true;
              $scope.successFlag = false;
            });
    	$scope.courseId = null;
    	$scope.date = null;
    	$scope.studentList = "";
    }
}]);

webControllers.controller('facultyCtrl', [ '$scope', '$http', '$location', function($scope, $http, $location) {
	//$scope.param = 5435;
	$scope.id = $location.path().split("/")[2];
}]);

webControllers.controller('facultyAttendanceViewCtrl', [ '$scope', '$http', '$location', function($scope, $http, $location) {
	$scope.val = $scope.param;
	$scope.id = $location.path().split("/")[2];
	$scope.successFlag = false;
        $scope.errorFlag = false;
        $http.get('/AttendancePortal/rest/facultyAttendanceViewDo', {params:{id:$scope.id}}).success(function(data) {
            console.log("Got data from server")
            $scope.response = data;
            if(data == false) {
            	$scope.errorFlag = true;
            	$scope.successFlag = false;
            } else {
            	$scope.successFlag = true;
            }
          }).error(function(){
              console.log("No reply from server");
              $scope.errorFlag = true;
              $scope.successFlag = false;
            });
    	$scope.id = null;
}]);

webControllers.controller('loginCtrl', [ '$scope', '$http', '$location', function($scope, $http, $location) {
	$scope.errorFlag = false;
	$scope.restCallData = false;
	$http.get('/AttendancePortal/rest/loginDo', {params:{username:$scope.username, password:$scope.password}}).success(function(data) {
		  console.log("Profile data"+ data);  
		  $scope.restCallData = data;
		});
    $scope.loginCheck = function(){
    	$scope.id = $scope.username;
        $scope.errorFlag = false;
        $http.get('/AttendancePortal/rest/loginDo', {params:{username:$scope.username, password:$scope.password}}).success(function(data) {
            console.log("Got data from server")
            $scope.response = data;
            if(data.isValid == false) {
            	$scope.errorFlag = true;
            } else {
            	if(data.accountType == "Admin") {
            		$location.path("admin");
            	} else if(data.accountType == "Faculty") {
            		$location.path("faculty/" + $scope.id);
            	} else {
            		$location.path("student/" + $scope.id);
            	}
            }
          }).error(function(){
              console.log("No reply from server for getAttributionData request");
              $scope.errorFlag = true;
            });

    	$scope.username = null;
    	$scope.password = null;
    }
}]);

webControllers.controller('studentCtrl', [ '$scope', '$http', '$location', function($scope, $http, $location) {
	$scope.id = $location.path().split("/")[2];
}]);

webControllers.controller('studentAttendanceViewCtrl', [ '$scope', '$http', '$location', function($scope, $http, $location) {
	$scope.val = $scope.param;
	$scope.id = $location.path().split("/")[2];
	$scope.successFlag = false;
        $scope.errorFlag = false;
        $http.get('/AttendancePortal/rest/studentAttendanceViewDo', {params:{id:$scope.id}}).success(function(data) {
            console.log("Got data from server")
            $scope.response = data;
            if(data == false) {
            	$scope.errorFlag = true;
            	$scope.successFlag = false;
            } else {
            	$scope.successFlag = true;
            }
          }).error(function(){
              console.log("No reply from server");
              $scope.errorFlag = true;
              $scope.successFlag = false;
            });
    	$scope.id = null;
}]);

webControllers.controller('aboutUsCtrl', [ '$scope', '$http', function($scope, $http) {
	
}]);

File: src/com/attendance/portal/Controllers.java
package com.attendance.portal;

import java.util.Arrays;
import java.util.LinkedList;

import javax.sql.DataSource;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseBody;

import com.google.gson.Gson;

@Controller
public class Controllers {
	
	@Autowired Gson gson;
	@Autowired DataSource dataSource;
	@Autowired Login loginHandler;
	
	@RequestMapping(value = "/sampleRestApi", method = RequestMethod.GET)
	public @ResponseBody
	String exampleFunction() {
		String text = "Server says hi!";
		return gson.toJson(text);
	}
	
	@RequestMapping(value = "/loginDo", method = RequestMethod.GET)
	public @ResponseBody
	String login(
			@RequestParam(value = "username", required = true) String username,
            @RequestParam(value = "password", required = true) String password) {
		System.out.println(username + "	" + password);
		Validation result;
		try{
			result = loginHandler.validate(dataSource, username, password);
		} catch(Exception e) {
			e.printStackTrace();
			result = new Validation(false, "Error");
		}
		return gson.toJson(result);
	}
	
	@RequestMapping(value = "/addFacultyDo", method = RequestMethod.GET)
	public @ResponseBody
	String addFaculty(
			@RequestParam(value = "firstName", required = true) String firstName,
            @RequestParam(value = "lastName", required = true) String lastName,
            @RequestParam(value = "id", required = true) String id,
            @RequestParam(value = "phoneNumber", required = true) String phoneNumber,
            @RequestParam(value = "password", required = true) String password) {
		System.out.println(firstName + "	" + lastName + "	" + id + "	" + phoneNumber + "	" + password);
		boolean result;
		try{
			result = Faculty.addFaculty(dataSource, id, lastName, firstName, phoneNumber, password);
			Faculty.viewFaculty(dataSource);
		} catch(Exception e) {
			e.printStackTrace();
			result = false;
		}
		return gson.toJson(result);
	}
	
	@RequestMapping(value = "/deleteFacultyDo", method = RequestMethod.GET)
	public @ResponseBody
	String deleteFaculty(
            @RequestParam(value = "id", required = true) String id) {
		System.out.println(id);
		boolean result;
		try{
			result = Faculty.deleteFaculty(dataSource, id);
			Faculty.viewFaculty(dataSource);
		} catch(Exception e) {
			e.printStackTrace();
			result = false;
		}
		return gson.toJson(result);
	}
	
	@RequestMapping(value = "/addStudentDo", method = RequestMethod.GET)
	public @ResponseBody
	String addStudent(
			@RequestParam(value = "firstName", required = true) String firstName,
            @RequestParam(value = "lastName", required = true) String lastName,
            @RequestParam(value = "fathersName", required = true) String fathersName,
            @RequestParam(value = "branch", required = true) String branch,
            @RequestParam(value = "year", required = true) String year,
            @RequestParam(value = "id", required = true) String id,
            @RequestParam(value = "phoneNumber", required = true) String phoneNumber,
            @RequestParam(value = "courses", required = true) String courses,
            @RequestParam(value = "password", required = true) String password) {
		System.out.println(firstName + "	" + lastName + "	" + id + "	" + phoneNumber + "	" + password);
		boolean result;
		try{
			result = Student.addStudent(dataSource, id, lastName, firstName, fathersName, Integer.parseInt(year), branch, phoneNumber, password);
			Student.viewStudent(dataSource);
			//Enroll student in courses
			LinkedList<String> courseList = (courses == null || courses.isEmpty()) ? new LinkedList<String>() : new LinkedList<String>(Arrays.asList(courses.split(" ")));
			Course_Student.addCourse_Student(dataSource, courseList, id);
		} catch(Exception e) {
			e.printStackTrace();
			result = false;
		}
		return gson.toJson(result);
	}
	
	@RequestMapping(value = "/deleteStudentDo", method = RequestMethod.GET)
	public @ResponseBody
	String deleteStudent(
            @RequestParam(value = "id", required = true) String id) {
		System.out.println(id);
		boolean result;
		try{
			result = Student.deleteStudent(dataSource, id);
			Student.viewStudent(dataSource);
		} catch(Exception e) {
			e.printStackTrace();
			result = false;
		}
		return gson.toJson(result);
	}
	
	@RequestMapping(value = "/enterAttendanceDo", method = RequestMethod.GET)
	public @ResponseBody
	String enterAttendance(
            @RequestParam(value = "courseId", required = true) String courseId,
            @RequestParam(value = "date", required = true) String date,
            @RequestParam(value = "studentList", required = true) String studentList) {
		System.out.println(courseId + "	" + date + "	" + studentList);
		boolean result;
		LinkedList<String> absentees = (studentList == null || studentList.isEmpty()) ? new LinkedList<String>() : new LinkedList<String>(Arrays.asList(studentList.split(" ")));
		try{
			result = EnterAttendance.enterAttendance(dataSource, courseId, date, absentees);
			EnterAttendance.viewAttendance(dataSource);
		} catch(Exception e) {
			e.printStackTrace();
			result = false;
		}
		return gson.toJson(result);
	}
	
	@RequestMapping(value = "/facultyAttendanceViewDo", method = RequestMethod.GET)
	public @ResponseBody
	String facultyAttendanceView(
            @RequestParam(value = "id", required = true) String id) {
		System.out.println(id);
		boolean result = false;
		try{
			//result = Student.deleteStudent(dataSource, id);
			//Student.viewStudent(dataSource);
		} catch(Exception e) {
			e.printStackTrace();
			result = false;
		}
		return gson.toJson(result);
	}
	
	@RequestMapping(value = "/studentAttendanceViewDo", method = RequestMethod.GET)
	public @ResponseBody
	String studentAttendanceView(
            @RequestParam(value = "id", required = true) String id) {
		System.out.println(id);
		boolean result = false;
		Attendance attendance;
		try{
			attendance = ViewAttendance.viewAttendance(dataSource, id);
			//Student.viewStudent(dataSource);
		} catch(Exception e) {
			e.printStackTrace();
			result = false;
			attendance = null;
		}
		return gson.toJson(attendance);
	}
}